### ISSUE TITLE
Markdown code chunks with specified language result in 'Malformed file'

### ISSUE THREAD

**Issue by lorenzwalthert** [jimhester/lintr#810]
An `.Rmd` file that contains a code block like this 
~~~
```yaml
- key: 
    value
``` 
~~~

Can't be linted: 
```r
lintr::lint('README.Rmd') # from r-lib/styler
#> Error: Malformed file!
```

Removing `yaml` makes the error go away, i.e.

~~~
```
- key: 
    value
``` 
~~~

Works fine. Since the former is part of the [GitHub flavoured markdown spec](https://guides.github.com/features/mastering-markdown/#), it should be recognised and handled properly.

Related: https://github.com/jimhester/lintr/issues/796, https://github.com/jimhester/lintr/issues/797.

**Comment by russHyde** [comment]
Thanks for posting this.
A similar bug occurs with html class tags
````
```{.my-class}
some block
```
````

I'll try to get this fixed this week.

**Comment by russHyde** [comment]
Apologies @lorenzwalthert , the "Malformed file" error appears to already have been fixed in https://github.com/jimhester/lintr/pull/575

I'll add some code blocks to the test.Rmd, so that these valid markdown blocks continue to be accepted by lintr.

This issue has only been fixed in the github version of lintr.
Will work on a release,

**Comment by lorenzwalthert** [comment]
Maybe it's worth checking if any of the related issues could also be closed for tust reason...
